src/pages/iou/IOUDetailsModal.js

```diff
@@ -0,0 +1,159 @@
+            // Temporarily placeholder, as the Venmo/Paypal issue will introduce settlement types.
+            settlementType: 'Elsewhere',
+        };
+
+        this.performIOUSettlement = this.performIOUSettlement.bind(this);
+    }
+
+    componentDidMount() {
+        // Fetch the iouReport without linking it to the chatReport. Else, the chatReport's iouReportID field would
+        // wrongly be pointed to an old settled iouReport, overwriting the iouReportID of the open iouReport.
+        fetchIOUReportByID(this.props.route.params.iouReportID, this.props.route.params.chatReportID);
+    }
+
+    performIOUSettlement() {
+        settleIOUReport({
+            chatReportID: this.props.route.params.chatReportID,
+            reportID: this.props.route.params.iouReportID,
+            paymentMethodType: this.state.settlementType,
+        });
+    }
+
+    render() {
+        const sessionEmail = lodashGet(this.props.session, 'email', null);
+        const reportIsLoading = _.isUndefined(this.props.iouReport);
+        return (
+            <ScreenWrapper>
+                <HeaderWithCloseButton
+                    title="Details"
+                    onCloseButtonPress={Navigation.dismissModal}
+                />
+                {reportIsLoading ? <ActivityIndicator color={themeColors.text} /> : (
+                    <View style={[styles.flex1, styles.justifyContentBetween]}>
+                        <ScrollView contentContainerStyle={{flexGrow: 1, paddingStart: 20, paddingEnd: 20}}>
+                            <ReportActionItemIOUPreview
+                                iou={this.props.iouReport}
+                                shouldHidePayButton
+                            />
+                            <IOUTransactions
+                                chatReportID={Number(this.props.route.params.chatReportID)}
+                                iouReportID={Number(this.props.route.params.iouReportID)}
+                                transactions={this.props.iouReport.transactions}
+                                hasOutstandingIOU={this.props.iouReport.hasOutstandingIOU}
+                            />
+                        </ScrollView>
+                        {(this.props.iouReport.hasOutstandingIOU
+                            && this.props.iouReport.managerEmail === sessionEmail && (
+                            <View style={styles.p5}>
+                                <ButtonWithLoader
+                                    text={this.props.translate('iou.settleElsewhere')}
+                                    isLoading={this.props.iou.loading}
+                                    onClick={this.performIOUSettlement}
+                                />
+                            </View>
+                        ))}
+                    </View>
+                )}
+            </ScreenWrapper>
+        );
+    }
+}
+
+IOUDetailsModal.propTypes = propTypes;
+IOUDetailsModal.displayName = 'IOUDetailsModal';
```

Comment:
Remove this as it is not necessary for class components.

Reply:
Oh nice, TIL.